tests/e2e/e2e_retry_policy_test.go

```diff
@@ -0,0 +1,284 @@
+					for _, n := range meshNs {
+						Expect(Td.CreateNs(n, nil)).To(Succeed())
+						Expect(Td.AddNsToMesh(true, n)).To(Succeed())
+					}
+
+					// Get simple pod definitions for the HTTP server
+					svcAccDef, podDef, svcDef, err := Td.SimplePodApp(
+						SimplePodAppDef{
+							PodName:   server,
+							Namespace: server,
+							Image:     "kennethreitz/httpbin",
+							Ports:     []int{80},
+							OS:        Td.ClusterOS,
+						})
+					Expect(err).NotTo(HaveOccurred())
+
+					_, err = Td.CreateServiceAccount(server, &svcAccDef)
+					Expect(err).NotTo(HaveOccurred())
+					_, err = Td.CreatePod(server, podDef)
+					Expect(err).NotTo(HaveOccurred())
+					serverSvc, err := Td.CreateService(server, svcDef)
+					Expect(err).NotTo(HaveOccurred())
+
+					Expect(Td.WaitForPodsRunningReady(server, 90*time.Second, 1, nil)).To(Succeed())
+
+					// Get simple Pod definitions for the source/client
+					svcAccDef, podDef, svcDef, err = Td.SimplePodApp(SimplePodAppDef{
+						PodName:   client,
+						Namespace: client,
+						Command:   []string{"sleep", "365d"},
+						Image:     "curlimages/curl",
+						Ports:     []int{80},
+						OS:        Td.ClusterOS,
+					})
+					Expect(err).NotTo(HaveOccurred())
+
+					clientSvcAcct, err := Td.CreateServiceAccount(client, &svcAccDef)
+					Expect(err).NotTo(HaveOccurred())
+					clientPod, err := Td.CreatePod(client, podDef)
+					Expect(err).NotTo(HaveOccurred())
+					_, err = Td.CreateService(client, svcDef)
+					Expect(err).NotTo(HaveOccurred())
+
+					Expect(Td.WaitForPodsRunningReady(client, 90*time.Second, 1, nil)).To(Succeed())
+
+					retry := &v1alpha1.Retry{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "retrypolicy",
+							Namespace: client,
+						},
+						Spec: v1alpha1.RetrySpec{
+							Source: v1alpha1.RetrySrcDstSpec{
+								Kind:      "ServiceAccount",
+								Name:      clientSvcAcct.Name,
+								Namespace: client,
+							},
+							Destinations: []v1alpha1.RetrySrcDstSpec{
+								{
+									Kind:      "Service",
+									Name:      serverSvc.Name,
+									Namespace: server,
+								},
+							},
+							RetryPolicy: v1alpha1.RetryPolicySpec{
+								RetryOn:                  "5xx",
+								PerTryTimeout:            &metav1.Duration{Duration: time.Duration(1 * time.Second)},
+								NumRetries:               &thresholdUintVal,
+								RetryBackoffBaseInterval: &metav1.Duration{Duration: time.Duration(5 * time.Second)},
+							},
+						},
+					}
+					_, err = Td.PolicyClient.PolicyV1alpha1().Retries(client).Create(context.TODO(), retry, metav1.CreateOptions{})
+					Expect(err).ToNot((HaveOccurred()))
+
+					req := HTTPRequestDef{
+						SourceNs:        client,
+						SourcePod:       clientPod.Name,
+						SourceContainer: podDef.GetName(),
+						Destination:     fmt.Sprintf("%s.%s.svc.cluster.local:80/status/503", serverSvc.Name, server),
+					}
+
+					By("A request that will be retried NumRetries times then succeed")
```

Comment:
I think the request eventually fails so should this line be the following?
```suggestion
					By("A request that will be retried NumRetries times and then fail")
```